### DISCUSSION TITLE
Call for translations

### DISCUSSION THREAD

**Discussion by gustavo-iniguez-goya** [evilsocket/opensnitch#484] [Development]
Hi!

@freddii , @ovari , @Danny3 : v1.4.0 is about to be released and I was wondering if German translation could be updated (freddii) and if any of you Danny or Ovari want to update the Romanian translation.

Thank you!

**Comment by ovari** [comment]
@gustavo-iniguez-goya can you please add your project to [Weblate](https://hosted.weblate.org/)?

Does Hungarian need updating?

Thank you

**Comment by ovari** [comment]
Please see https://github.com/gaphor/gaphor/issues/937 for more details. Thank you

**Comment by freddii** [comment]
yes would be nice to be able to edit it on weblate.

**Reply by ovari** [reply]
Seems to be possible now.

**Comment by gustavo-iniguez-goya** [comment]
https://hosted.weblate.org/projects/opensnitch/gui/

I asked for "libre hosting" as they call it.

**Reply by ovari** [reply]
https://github.com/evilsocket/opensnitch/pull/486

**Reply by Herve5** [reply]
I just created a French translation on this Weblate server, but I don't know how to install it here afterwards. 
Moreover, as Weblate conditions of use impose that my email is sold for ads, I did not give it, instead I used one that I just don't read -so for any comments, please reply here ;-)
I hereby donate this translation package to the OpenSnitch project, in case it helps.

**Reply by gustavo-iniguez-goya** [reply]
Thank you very much @Herve5 ! and sorry for the email thing ... I read their policy and didn't find nothing about the ads , probably didn't pay attention (maybe the data sharing with 3rd party sources?).

Anyway, the instructions to generate the translation are here: https://github.com/evilsocket/opensnitch/tree/master/ui/i18n

Basically:
```
$ cd opensnitch/ui/i18n
$ vim opensnitch_i18n.pro <-- add fr_FR lines as stated in the link above
$ pylupdate5 opensnitch_i18n.pro
$ lrelease locales/fr-FR/opensnitch-fr_FR.ts -qm locales/fr-FR/opensnitch-fr_FR.qm
$ sudo mkdir /usr/local/lib/python3.9/dist-packages/opensnitch/i18n/
$ sudo cp locales/fr-FR/opensnitch-fr_FR.qm /usr/local/lib/python3.9/dist-packages/opensnitch/i18n/fr_FR/
```
[deleted comment referring to Durations]

These strings must be without spaces:
https://hosted.weblate.org/translate/opensnitch/gui/fr/?checksum=5777839e16d3ff25&sort_by=-priority,position
https://hosted.weblate.org/translate/opensnitch/gui/fr/?checksum=ff3cda5ec09535e0&sort_by=-priority,position

Sorry for the confusion, I'll set those strings as not translatable in the future.

**Comment by ovari** [comment]
@gustavo-iniguez-goya please allow  "30s", "5m", "15m" and "1h" to be translated.
* `30s` is `30 másodperc` in Hungarian. Could a non-breaking space be used? If it needs to be shorter, without a space is could be `30mp`
* Likewise for `5m` is `5 perc` and if it needs to be shorter `5p`
* Similarly `1h` is `1 óra` and if this needs to be shorter `1ó`

Perhaps you could make a comment with the key for strings that must be not contain spaces?

What do you think?

Thank you

**Reply by gustavo-iniguez-goya** [reply]
Ok, I fixed it here https://github.com/evilsocket/opensnitch/commit/18f0c9133d3fca284188a2c51005730afde02139

I was reviewing the translations and realized that the japanese durations were also translated. Hungarian translation should also work now.

https://github.com/evilsocket/opensnitch/commit/18f0c9133d3fca284188a2c51005730afde02139

@Herve5 , you don't need to change the durations now, leave them as you translated them.

> Perhaps you could make a comment with the key for strings that must be not contain spaces?

There is already a comment, using linguist:

![image](https://user-images.githubusercontent.com/2742953/131592781-b27cf2db-a492-4389-b1d7-6573355250de.png)

weblate doesn't seem to interpret it correctly though.

**Reply by Herve5** [reply]
Silly question (I am overloaded now) : how do I recuperate the french string package from Weblate? I don't know anything at all apart translating... (I'm going to correct the white spaces in the '10s', '30s' etc. translations)

**Reply by gustavo-iniguez-goya** [reply]
From https://hosted.weblate.org/projects/opensnitch/gui/fr/ click on the menu Files -> Download translation

**Reply by gustavo-iniguez-goya** [reply]
@Herve5 I can add the translation for you if you're too busy, np.

**Reply by Herve5** [reply]
Indeed Gustavo this would be very helpful : I'm still on my way back home and will start working tomorrow with one day's off of rush... and I understand the next OpenSnitch version now is imminent, so it'd be nice with the french translation in...

I promiss I'll do my best for the next updates!

Thank you!

Hervé

**Comment by ovari** [comment]
@Herve5 can you please provide a link to where they sell email addresses as we couldn't find it? Thank you

**Reply by Herve5** [reply]
Indeed as you saw initially, the sentence is something like "you agree we"ll communicate your address to our partners", which doesn't say "ad" but allows everything...
This sentence only happens AFTER you have given your email actually, within the registration process. So I didn't finalize the process but in fact I did send them my personal email in the beginning. I just hope they deleted it because I didn't finalize. And then aterwards I registered with a trash email.

**Reply by ovari** [reply]
Perhaps they use your email address with third-party sign-in services, like GitHub, to make it easy for you to sign-in.

**Comment by Danny3** [comment]
Sorry for the late reply, I haven't entered Github in quite a while.

Thanks for the notification and for using an online translation service !
I have started contributing to the romanian translation.

**Comment by BobSquarePants** [comment]
Hi everyone,

I would like to give my modest help for OpenSnitch. 

is there a way to be notified if a language fall under 100%
![lJlGfkw](https://user-images.githubusercontent.com/23012674/151351247-41987545-7ac0-472a-a145-ef7da488f552.png)


Thanks

**Reply by gustavo-iniguez-goya** [reply]
No idea @BobSquarePants .

By the way, I've tried out to sync translations with latest from this git, but it has failed. I've tried to fix it but failed.

**Comment by jermanuts** [comment]
Hey @gustavo-iniguez-goya on Weblate, should we translate main branch? and why do we have components for old releases?

The Weblate translation progress list in https://github.com/evilsocket/opensnitch?tab=readme-ov-file#translating needs a refresh as it is only showing 2 languages.

**Reply by jermanuts** [reply]
I opened a PR to fix it https://github.com/evilsocket/opensnitch/pull/1289 thanks for working on opensnitch btw ❤️ @gustavo-iniguez-goya

**Comment by pablofrelo** [comment]
Hi! I'd like to add a Polish (pl_PL) translation of the GUI. I'm a native
Polish speaker and a daily OpenSnitch user.

Before I start: Weblate is currently closed for maintenance and seems to
track the 1.4.0 branch. Which way do you prefer for new translations —
Weblate (once it's back), or a pull request with
ui/i18n/locales/pl_PL/opensnitch-pl_PL.ts generated from the current
master?

I may use AI assistance for a first draft, but I'll review every string
myself.

**Comment by pablofrelo** [comment]
Opened #1673  with a complete pl_PL translation against master.
